pkg/karmadactl/join.go

```diff
@@ -62,25 +61,22 @@ func NewCmdJoin(cmdOut io.Writer, karmadaConfig KarmadaConfig, cmdStr string) *c
 	opts := CommandJoinOption{}
 
 	cmd := &cobra.Command{
-		Use:     "join CLUSTER_NAME --cluster-kubeconfig=<KUBECONFIG>",
-		Short:   joinShort,
-		Long:    joinLong,
-		Example: fmt.Sprintf(joinExample, cmdStr),
-		Run: func(cmd *cobra.Command, args []string) {
-			// Set default values
-			err := opts.Complete(args)
-			if err != nil {
-				klog.Fatalf("Error: %v", err)
+		Use:          "join CLUSTER_NAME --cluster-kubeconfig=<KUBECONFIG>",
+		Short:        joinShort,
+		Long:         joinLong,
+		Example:      fmt.Sprintf(joinExample, cmdStr),
+		SilenceUsage: true,
```

Comment:
I don't know when we should `SlilenceUsage`, could you please give an example for this?

Reply:
If we don`t set SilenceUsage, it not only report error, but also print Usage, the context is too much to find the useful information. The situation is shown below:

```
[root@master67 karmada]# karmadactl join
Error: cluster name is required
Usage:
  karmadactl join CLUSTER_NAME --cluster-kubeconfig=<KUBECONFIG> [flags]

Examples:

karmadactl join member1 --cluster-kubeconfig=$HOME/.kube/members.config


Flags:
      --cluster-context string      Context name of cluster in kubeconfig. Only works when there are multiple contexts in the kubeconfig.
      --cluster-kubeconfig string   Path of the cluster's kubeconfig.
      --cluster-namespace string    Namespace in the control plane where member cluster are stored. (default "karmada-cluster")
      --dry-run                     Run the command in dry-run mode, without making any server requests.
  -h, --help                        help for join
      --karmada-context string      Name of the cluster context in control plane kubeconfig file.

Global Flags:
      --add-dir-header                   If true, adds the file directory to the header of the log messages
      --alsologtostderr                  log to standard error as well as files
      --kubeconfig string                Paths to a kubeconfig. Only required if out-of-cluster.
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --log-file string                  If non-empty, use this log file
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
      --log-flush-frequency duration     Maximum number of seconds between log flushes (default 5s)
      --logtostderr                      log to standard error instead of files (default true)
      --one-output                       If true, only write logs to their native severity level (vs also writing to each lower severity level)
      --skip-headers                     If true, avoid header prefixes in the log messages
      --skip-log-headers                 If true, avoid headers when opening log files
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
  -v, --v Level                          number for the log level verbosity
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
```
`karmadactl join -h` also get these information. Printing usage every time an error is reported is a bit redundant

Reply:
Do you mean we should set `SlilenceUsage==true` to all commands?
The usage only dumped by `karmadactl [xxx] -h`?

Reply:
Yes, if necessary, we can customize brief hint for a sub command, but needn`t to print default Usage that is too long.

Reply:
OK.